### PR TITLE
Update docs for Go client

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolongran <xiaolongran@tencent.com>

### Motivation

With the iteration of the Go client version, we have introduced many new interfaces and parameters, but the relevant documents have not been updated, resulting in inconsistencies between the documents and the code. This pr will focus on solving this problem and sort out all the parameters and interface lists related to `client`, `producer`, `consumer`and `reader` in the current Go client. Ensure that the documentation and code are consistent.

### Modifications

- Add `Schema` docs and code example for producer and consumer
- update some docs

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### CODE REVIEW

File: site2/docs/client-libraries-go.md [hunk 1]
```diff
@@ -117,6 +117,10 @@ The following configurable parameters are available for Pulsar clients:
 | TLSTrustCertsFilePath | Set the path to the trusted TLS certificate file | |
 | TLSAllowInsecureConnection | Configure whether the Pulsar client accept untrusted TLS certificate from broker | false |
 | TLSValidateHostname | Configure whether the Pulsar client verify the validity of the host name from broker | false |
+| ListenerName | Configure the net model for vpc user to connect the pulsar broker |  |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:120]:
```suggestion
| ListenerName | Configure the net model for VPC users to connect to the Pulsar broker |  |
```

File: site2/docs/client-libraries-go.md [hunk 2]
```diff
@@ -117,6 +117,10 @@ The following configurable parameters are available for Pulsar clients:
...
+| MaxConnectionsPerBroker | Max number of connections to a single broker that will kept in the pool | 1 |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:121]:
```suggestion
| MaxConnectionsPerBroker | Max number of connections to a single broker that is kept in the pool | 1 |
```

File: site2/docs/client-libraries-go.md [hunk 3]
```diff
@@ -260,13 +310,21 @@ canc()
 | Topic | Topic specify the topic this consumer will subscribe to. This argument is required when constructing the reader. | |
 | Name | Name specify a name for the producer. If not assigned, the system will generate a globally unique name which can be access with Producer.ProducerName(). | | 
 | Properties | Properties attach a set of application defined properties to the producer This properties will be visible in the topic stats | |
+| SendTimeout | SendTimeout set the timeout for a message that not be acknowledged by server since sent | 30s |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:313]:
```suggestion
| SendTimeout | SendTimeout set the timeout for a message that is not acknowledged by the server  | 30s |
```

File: site2/docs/client-libraries-go.md [hunk 4]
```diff
@@ -260,13 +310,21 @@ canc()
...
 | HashingScheme | HashingScheme change the `HashingScheme` used to chose the partition on where to publish a particular message. | JavaStringHash |
 | CompressionType | CompressionType set the compression type for the producer. | not compressed | 
+| CompressionLevel | Define the desired compression level. Options: Default, Faster and Better | Default  | 
 | MessageRouter | MessageRouter set a custom message routing policy by passing an implementation of MessageRouter | |
 | DisableBatching | DisableBatching control whether automatic batching of messages is enabled for the producer. | false |
 | BatchingMaxPublishDelay | BatchingMaxPublishDelay set the time period within which the messages sent will be batched | 1ms |
 | BatchingMaxMessages | BatchingMaxMessages set the maximum number of messages permitted in a batch. | 1000 | 
+| BatchingMaxSize | BatchingMaxSize sets the maximum number of bytes permitted in a batch. | 128KB | 
+| Schema |  Schema set a custom schema type by passing an implementation of `Schema` | bytes[] | 
+| Interceptors | A chain of interceptors, These interceptors will be called at some points defined in ProducerInterceptor interface. | None |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:325]:
```suggestion
| Interceptors | A chain of interceptors. These interceptors are called at some points defined in the `ProducerInterceptor` interface. | None | 
```

File: site2/docs/client-libraries-go.md [hunk 5]
```diff
@@ -260,13 +310,21 @@ canc()
...
 | MessageRouter | MessageRouter set a custom message routing policy by passing an implementation of MessageRouter | |
 | DisableBatching | DisableBatching control whether automatic batching of messages is enabled for the producer. | false |
 | BatchingMaxPublishDelay | BatchingMaxPublishDelay set the time period within which the messages sent will be batched | 1ms |
 | BatchingMaxMessages | BatchingMaxMessages set the maximum number of messages permitted in a batch. | 1000 | 
+| BatchingMaxSize | BatchingMaxSize sets the maximum number of bytes permitted in a batch. | 128KB | 
+| Schema |  Schema set a custom schema type by passing an implementation of `Schema` | bytes[] | 
+| Interceptors | A chain of interceptors, These interceptors will be called at some points defined in ProducerInterceptor interface. | None | 
+| MaxReconnectToBroker | MaxReconnectToBroker set the maximum retry number of reconnectToBroker | ultimate | 
+| BatcherBuilderType | BatcherBuilderType sets the batch builder type, this will be used to create batch container when batching is enabled. Options: DefaultBatchBuilder and KeyBasedBatchBuilder | DefaultBatchBuilder |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:327]:
```suggestion
| BatcherBuilderType | BatcherBuilderType sets the batch builder type. This is used to create a batch container when batching is enabled. Options: DefaultBatchBuilder and KeyBasedBatchBuilder | DefaultBatchBuilder | 
```

File: site2/docs/client-libraries-go.md [hunk 6]
```diff
@@ -309,17 +367,20 @@ Method | Description | Return type
 `Subscription()` | Returns the consumer's subscription name | `string`
 `Unsubcribe()` | Unsubscribes the consumer from the assigned topic. Throws an error if the unsubscribe operation is somehow unsuccessful. | `error`
 `Receive(context.Context)` | Receives a single message from the topic. This method blocks until a message is available. | `(Message, error)`
+`Chan()` | Chan returns a channel to consume messages from. | `<-chan ConsumerMessage`
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:370]:
```suggestion
`Chan()` | Chan returns a channel from which to consume messages. | `<-chan ConsumerMessage`
```

File: site2/docs/client-libraries-go.md [hunk 7]
```diff
@@ -478,6 +584,11 @@ if err != nil {
 | NackRedeliveryDelay | The delay after which to redeliver the messages that failed to be processed | 1min |
 | ReadCompacted | If enabled, the consumer will read messages from the compacted topic rather than reading the full message backlog of the topic | false |
 | ReplicateSubscriptionState | Mark the subscription as replicated to keep it in sync across clusters | false |
+| KeySharedPolicy | Configuration for Key Shared consumer policy. |  |
+| RetryEnable | Auto retry send messages to default filled DLQPolicy topics | false |
+| Interceptors | A chain of interceptors, These interceptors will be called at some points defined in ConsumerInterceptor interface. |  |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:589]:
```suggestion
| Interceptors | A chain of interceptors. These interceptors are called at some points defined in the `ConsumerInterceptor` interface. |  |
```

File: site2/docs/client-libraries-go.md [hunk 8]
```diff
@@ -504,6 +615,8 @@ Method | Description | Return type
 `Next(context.Context)` | Receives the next message on the topic (analogous to the `Receive` method for [consumers](#consumer-operations)). This method blocks until a message is available. | `(Message, error)`
 `HasNext()` | Check if there is any message available to read from the current position| (bool, error)
 `Close()` | Closes the reader, disabling its ability to receive messages from the broker | `error`
+`Seek(MessageID)` | Reset the subscription associated with this reader to a specific message id | `error`
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:618]:
```suggestion
`Seek(MessageID)` | Reset the subscription associated with this reader to a specific message ID | `error`
```

**Review by Anonymitaet — COMMENTED** [review]
Many thanks for your contribution. I’ve left some comments, PTAL, thanks.

**Comment by wolfstudy** [comment]
@Anonymitaet Thanks yu, PTAL again, thanks.

**Review by Anonymitaet — APPROVED** [review]

**Review by merlimat — APPROVED** [review]

**Pull request merged**